gcs/innards/state.hh

```diff
@@ -516,6 +519,9 @@ namespace gcs::innards
         [[nodiscard]] auto current() -> CurrentState;
 
         ///@}
+
```

Comment:
Is it worth having a very cheap wrapper struct rather than using `unsigned long` directly? I kind of dislike having naked integers floating around in an API because it's a pain to find and change them later on.

Reply:
As in, something like:
```c++
struct DFAStateID { 
     unsigned long raw_value;
}
```

and then just accessing `.raw_value` everywhere?

Although, sometimes the long is representing a state ID and sometimes it's representing a position in the sequence of variables... not sure what's best to do in the latter case?

Or I could replace everything with `gcs::Integer`?

Reply:
I think this comment might have moved slightly after some commits, but it refers to the return value of `add_constraint_state` and the value passed to `get_constraint_state`. I feel like these should be a `struct ConstraintStateHandle { unsigned long index };` just so the user can't easily pass in something else as an argument.

Reply:
Ahh I see. Okay yes that's much less painful to change